framework/include/meshmodifiers/ElementSubdomainModifierBase.h

```diff
@@ -136,13 +198,63 @@ class ElementSubdomainModifierBase : public ElementUserObject
   std::unordered_set<dof_id_type> _reinitialized_elems;
   /// Range of reinitialized elements
   std::unique_ptr<ConstElemRange> _reinitialized_elem_range;
-  /// Range of reinitialized elements on the displaced mesh
-  std::unique_ptr<ConstElemRange> _reinitialized_displaced_elem_range;
+
+  /// Semi-local Reinitialized elements:
+  /// Ghosted and local reinitialized elements
+  std::unordered_set<dof_id_type> _semi_local_reinitialized_elems;
 
   /// Reinitialized nodes
   std::unordered_set<dof_id_type> _reinitialized_nodes;
+  /// Range of reinitialized nodes
+  std::unique_ptr<ConstNodeRange> _reinitialized_node_range;
   /// Range of reinitialized boundary nodes
   std::unique_ptr<ConstBndNodeRange> _reinitialized_bnd_node_range;
-  /// Range of reinitialized boundary nodes on the displaced mesh
-  std::unique_ptr<ConstBndNodeRange> _reinitialized_displaced_bnd_node_range;
+
+  /// A map from variable name to a pair of:
+  /// (1) a vector of DOF IDs associated with non-reinitialized nodes on reinitialized elements, and
+  /// (2) the corresponding solution values at those DOFs.
+  /// This map is used to preserve solution data for variables that should not be reinitialized
+  /// even though they reside on reinitialized elements.
+  std::map<VariableName, std::pair<std::vector<dof_id_type>, std::vector<Number>>>
+      _overridden_values_on_reinit_elems;
+
+  /// The strategy used to apply IC on newly activated nodes
```

Comment:
```suggestion
  /// The strategy used to apply IC on newly activated elements
```
nodes? does this only work for first lagrange?

Reply:
It is a function space independent implementation. I just fixed this.